src/BulkWriter/Pipelines/Tasks/TaskBase.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Threading.Tasks;
+
+namespace BulkWriter.Pipelines.Tasks
+{
+    internal abstract class TaskBase
+    {
+        protected readonly TaskFactory TaskFactory;
+
+        protected TaskBase(TaskFactory taskFactory)
+        {
+            this.TaskFactory = taskFactory;
+        }
+
+        public abstract void Run();
+    }
+
+    internal abstract class TaskBase<T> : TaskBase
+    {
+        protected TaskBase(TaskFactory taskFactory, BlockingCollection<T> consumingCollection) : base(taskFactory)
+        {
+            ConsumingCollection = consumingCollection;
+        }
+
+        protected BlockingCollection<T> ConsumingCollection { get; }
```

Comment:
Would renaming this help add clarity?  Looking through the pipeline code it seems like this could be `OutputCollection` but perhaps I'm misunderstanding how it's being used.